src/test/gcp/cloudfunctions.spec.ts

```diff
@@ -258,4 +258,50 @@ describe("cloudfunctions", () => {
       });
     });
   });
+
+  describe("generateIamPolicy", () => {
+    it("should throw error on empty service account string", () => {
+      expect(() => cloudfunctions.generateIamPolicy("project", "")).to.throw();
+    });
+
+    it("should generate the public policy", () => {
+      const policy = cloudfunctions.generateIamPolicy("project", "public");
+      expect(policy.bindings.length).to.eq(1);
+      expect(policy.bindings[0].members.length).to.eq(1);
+      expect(policy.bindings[0].members[0]).to.eq("allUsers");
+    });
+
+    it("should generate the private policy", () => {
+      const policy = cloudfunctions.generateIamPolicy("project", "private");
+      expect(policy.bindings.length).to.eq(1);
+      expect(policy.bindings[0].members.length).to.eq(0);
+    });
+
+    it("should generate a policy for a single service account", () => {
```

Comment:
Can you add a test for passing a full service account (service-account@proj.iam.gserviceaccount.com) as well?

Reply:
Added a check and tests for using a full service account in the invoker